internal/state/graph/graph.go

```diff
@@ -35,14 +36,32 @@ type Graph struct {
 	IgnoredGateways map[types.NamespacedName]*v1beta1.Gateway
 	// Routes holds Route resources.
 	Routes map[types.NamespacedName]*Route
+	// Secrets includes Secrets used by Gateway Listeners, including invalid ones.
+	// It is different from other similar maps, because it includes entries for Secrets that do not exist
+	// in the cluster. We need such entries so that we can query the Graph to determine if a Secret is referenced
+	// by the Gateway, including the case when the Secret is newly created.
+	Secrets map[types.NamespacedName]*Secret
+}
+
+// IsReferenced returns true if the Graph references the resource.
+func (g *Graph) IsReferenced(resourceType client.Object, nsname types.NamespacedName) bool {
+	// FIMXE(pleshakov): For now, only works with Secrets.
```

Comment:
Rather than the FIXME, can we just add a simple switch statement that can be enhanced in the future, and then create a tech debt story to remove the capturer, and include any necessary details in there?

Reply:
yep, I will update with an issue for the FIXME. I didn't create it pending the review - since a review can potentially can end up with a different design that doesn't need any FIXMEs.

Reply:
https://github.com/nginxinc/nginx-kubernetes-gateway/issues/824  (I will add the link to the code too)